packages/flutter_tools/test/general.shard/ios/xcodeproj_test.dart

```diff
@@ -129,170 +188,162 @@ void main() {
-    when(processManager.runSync(<String>[xcodebuild, '-version']))
-        .thenReturn(ProcessResult(1, 0, 'Xcode 8.3.3\nBuild version 8E3004b', ''));
+    fakeProcessManager.addCommand(const FakeCommand(
+      command: <String>[xcodebuild, '-version'],
+      stdout: 'Xcode 8.3.3\nBuild version 8E3004b',
+    ));
 
     expect(xcodeProjectInterpreter.isInstalled, isTrue);
+    expect(fakeProcessManager.hasRemainingExpectations, isFalse);
   });
 
   testWithoutContext('xcodebuild build settings is empty when xcodebuild failed to get the build settings', () async {
     platform.environment = Map<String, String>.unmodifiable(<String, String>{});
-    when(processManager.runSync(
-              argThat(contains(xcodebuild)),
-              workingDirectory: anyNamed('workingDirectory'),
-              environment: anyNamed('environment')))
-        .thenReturn(ProcessResult(0, 1, '', ''));
 
-    expect(await xcodeProjectInterpreter.getBuildSettings('', ''), const <String, String>{});
-  });
-
-  testWithoutContext('xcodebuild build settings flakes', () async {
-    const Duration delay = Duration(seconds: 1);
-    processManager.processFactory = mocks.flakyProcessFactory(
-      flakes: 1,
-      delay: delay + const Duration(seconds: 1),
-    );
-    platform.environment = Map<String, String>.unmodifiable(<String, String>{});
+    fakeProcessManager.addCommand(const FakeCommand(
+      command: <String>[
+        '/usr/bin/xcodebuild',
+        '-project',
+        '/',
+        '-target',
+        '',
+        '-showBuildSettings'
+      ],
+      exitCode: 1,
+    ));
 
-    expect(await xcodeProjectInterpreter.getBuildSettings(
-                '', '', timeout: delay),
-            const <String, String>{});
-    // build settings times out and is killed once, then succeeds.
-    verify(processManager.killPid(any)).called(1);
-    // The verbose logs should tell us something timed out.
-    expect(logger.traceText, contains('timed out'));
+    expect(await xcodeProjectInterpreter.getBuildSettings('', ''), const <String, String>{});
+    expect(fakeProcessManager.hasRemainingExpectations, isFalse);
   });
 
   testWithoutContext('xcodebuild build settings contains Flutter Xcode environment variables', () async {
     platform.environment = Map<String, String>.unmodifiable(<String, String>{
       'FLUTTER_XCODE_CODE_SIGN_STYLE': 'Manual',
       'FLUTTER_XCODE_ARCHS': 'arm64'
     });
-    when(processManager.runSync(<String>[
-      xcodebuild,
-      '-project',
-      platform.pathSeparator,
-      '-target',
-      '',
-      '-showBuildSettings',
-      'CODE_SIGN_STYLE=Manual',
-      'ARCHS=arm64'
-    ],
-      workingDirectory: anyNamed('workingDirectory'),
-      environment: anyNamed('environment')))
-      .thenReturn(ProcessResult(1, 0, '', ''));
+    fakeProcessManager.addCommand(FakeCommand(
+      command: <String>[
+        xcodebuild,
+        '-project',
+        fileSystem.path.separator,
```

Comment:
This was the test that should have failed but didn't.

`platform.pathSeparator` is null, `fileSystem.path.separator` is `/` but this processManager `runSync` was never being verified.